apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SimpleSubscriber.cs

```diff
@@ -805,13 +803,15 @@ private void QueueLeaseExtensions(IEnumerable<string> ackIds)
                         acks = _ackQueue.Take(_maxAckExtendCount).ToList();
                         extends = _extendQueue.Take(_maxAckExtendCount - acks.Count).ToList();
                     }
-                    // Send acks/extensions
-                    await _taskHelper.ConfigureAwait(pull.WriteAsync(new StreamingPullRequest
+                    var req = new StreamingPullRequest
                     {
                         AckIds = { acks },
-                        ModifyDeadlineAckIds = { extends },
-                        ModifyDeadlineSeconds = { Enumerable.Repeat(_modifyDeadlineSeconds, extends.Count) },
-                    }));
+                        // Don't lease-extend any msg that is also being acked.
+                        ModifyDeadlineAckIds = { extends.Except(acks) }
```

Comment:
Might this be expensive? Is there an alternative approach we should consider?

Reply:
I've considered various alternatives, but I think this is the best compromise.
The size of both `acks` and `extends` are limited to 10,000 so this will never be too expensive.